test/tests_for_sdkmr/ee2_SDKMethodRunner_test_EE2Runjob_test.py

```diff
@@ -247,6 +247,189 @@ def test_run_job(self, rq_mock, condor_mock):
         job_id = runner.run_job(params=job)
         print(f"Job id is {job_id} ")
 
+    @requests_mock.Mocker()
+    @patch("lib.execution_engine2.utils.Condor.Condor", autospec=True)
+    def test_retry_job_multiple(self, rq_mock, condor_mock):
+        # 1. Run the job
+        rq_mock.add_matcher(
+            run_job_adapter(
+                ws_perms_info={"user_id": self.user_id, "ws_perms": {self.ws_id: "a"}}
+            )
+        )
+        runner = self.getRunner()
+        runner.get_condor = MagicMock(return_value=condor_mock)
+        runner.workspace.get_object_info3 = MagicMock(return_value={"paths": []})
+        job = get_example_job_as_dict(
+            user=self.user_id, wsid=self.ws_id, source_ws_objects=[]
+        )
+        si = SubmissionInfo(clusterid="test", submit=job, error=None)
+        condor_mock.run_job = MagicMock(return_value=si)
+        parent_job_id1 = runner.run_job(params=job)
+        parent_job_id2 = runner.run_job(params=job)
+
+        parent_job_id1 = runner.run_job(params=job)
+        parent_job_id2 = runner.run_job(params=job)
+        runner.update_job_status(job_id=parent_job_id1, status=Status.terminated.value)
+        runner.update_job_status(job_id=parent_job_id2, status=Status.error.value)
+
+        # 2. Retry the jobs with a fake input
+        errmsg = (
+            "'123' is not a valid ObjectId, it must be a 12-byte input or a 24-character "
+            "hex string"
+        )
+        with self.assertRaisesRegexp(RetryFailureException, errmsg):
+            runner.retry_multiple(job_ids=[parent_job_id1, 123])
+
+        # 2. Actually retry the jobs
+        retry_candidates = (
+            parent_job_id1,
+            parent_job_id2,
+            parent_job_id1,
+            parent_job_id2,
```

Comment:
I could see wanting to re-run the same job with different resources to learn how much resources the dataset needs or the app needs to then apply that resource request to other jobs, but we don't support resource requests in retries right now so, so yea we could add that error message. If we supplied two different job ids of jobs that already ran, they end up running the retry_parent anyway, so  could this be handled when we handle preventing multiple in-flight retries?

Reply:
An alternative approach could be something like:
```
seen = {}
result = []
for j in jobs:
     if j in seen:
        result.append(seen[j])
     result.append(self._retry(j))
```
If resource requirements were added the if clause would just need to be modified to check whether the resource requirements are different as well.

Reply:
How should we proceed? Should we do anything now or let it be the way it is now?

Reply:
Well,
1) we could allow retrying the same job at the same time in a batch. This is a waste of resources and is likely a programming bug, but _probably_ isn't going to happen very often. If the client programmer doesn't like the behavior all they have to do is not submit duplicate jobs.
2) We could throw an error if there are identical jobs in the retry list. This wouldn't catch the case of two different children of the same parent being retried, which would have the same effect of wasting resources - but in that case IMO we probably shouldn't error out - that's just making things hard for the client. In that case we could either let things roll or do the duplicate detection scheme in my prior post to avoid wasting resources.
3) We could do the duplicate detection scheme above, but that also wouldn't catch the case of two children of the same parent in the batch. For that the resolution of the real job retry list (e.g. job IDs post recursion) would need to happen before the deduplication.

For 2) or 3) it could be done for this PR or added to the ADR.

After further thought I'm ok with any of the 5 choices, other than if we do 1) we should document it in the spec file. Otherwise up to you.

Reply:
I've pinged @ialarmedalien for her opinion again, since she's going to have to code against the endpoint, not me, so her opinion has more weight than mine.

Reply:
Re: Boris's first comment: we need to nail down what makes a job part of a batch -- IMO job params and environment params (resource requirements) should not be changed within a batch, and retries exactly replicate the params and conditions in the original job. Change the resource requirements or job params => it's part of a new batch.

Reply:
@ialarmedalien I'm not sure I understand the rationale - for example, if a job in a batch fails because it doesn't have enough memory, I don't see why increasing the memory on the retry should mean that the job should be removed from the batch.

This particular issue is more or less moot though AFAICT - for one thing, there's no way to manually specify job requirements on retries currently, and for another, if the clientgroups in the catalog are altered between the original run and the retry run, the job requirements will change anyway.

Reply:
> job params... should not be changed within a batch

Reply:
I think if the job params for a particular job is changed, it is no longer that job anymore, but a new job, and it's not really a "retry", but more of a new experiment/job. We currently do not support changing params for a job within a retry either.

Reply:
Adding a simple check for the same ids and throwing an error right now.. We can add this issue to the ADR..

Reply:
Agree with that

Reply:
Boris's interpretation is what I meant.

re: resource requirements: if they can't be changed at the moment, it doesn't matter for the purposes of this discussion.

Reply:
From the perspective of the FE, jobs are going to be grouped based on the job ID in the initial run, and there is no way to display multiple concurrent reruns of the same job in the current table format (well, no way to do it without things getting ugly--fast). Resolving retry requests to the appropriate `retry_parent` and deduplicating to prevent wasted work would be my strong preference.

Reply:
@ialarmedalien the thrust of this issue was around what should happen if batch retry is called with duplicate job IDs, or job IDs that resolve to the same parent ID - do you have any opinion there?

Reply:
I got in there ten seconds before you asked!

Reply:
:P Ok, next question - can this go in the ADR and wait a bit or no?

Reply:
I think get this PR out (assuming you have no other code-related comments), and then make refinements.